src/serialize.ts

```diff
@@ -32,6 +30,7 @@ function getCompileOptions(
     outputFormat: 'function-body',
     // Disable the importSource option for RSC to ensure there's no `useMDXComponents` implemented.
     providerImportSource: rsc ? undefined : '@mdx-js/react',
+    development: process.env.NODE_ENV === 'development',
```

Comment:
I couldnt get the runtime switching to work in the integration tests without adding this flag.  The mdx-js docs add it --> https://mdxjs.com/packages/mdx/#jsxdev